cmd/load-runner/runner.go

```diff
@@ -169,6 +219,48 @@ func (r *LoadTestRunner) monitorMetrics(ctx context.Context) {
 				fields[i] = logger.NewField(m.Key, m.Value)
 			}
 			r.logger.Infon("Load test metrics", fields...)
+
+			// Record metrics for file output
+			r.recordMetrics(metrics)
 		}
 	}
 }
+
+// recordMetrics adds a new metrics record to the runner's metrics data
+func (r *LoadTestRunner) recordMetrics(metrics []metrics.MetricsResponse) {
+	r.metricsMutex.Lock()
+	defer r.metricsMutex.Unlock()
+
+	r.metricsData = append(r.metricsData, metricsRecord{
+		Timestamp: time.Now(),
+		Metrics:   metrics,
+	})
+}
+
+func (r *LoadTestRunner) writeMetricsToFile() error {
+	r.metricsMutex.Lock()
+	defer r.metricsMutex.Unlock()
+
+	if len(r.metricsData) == 0 {
+		return nil
+	}
+
+	// Create directory if it doesn't exist
+	dir := "metrics_reports"
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("failed to create metrics directory: %w", err)
+	}
+
+	filePath := filepath.Join(dir, r.metricsFile)
```

Comment:
One question. Given that you're writing the whole thing at the end, wouldn't it be simpler to stop recording metrics and then call `writeMetricsToFile` without a mutex lock?